src/features/Send/common/hooks.ts

```diff
@@ -0,0 +1,50 @@
+import {useSelectedWallet} from '../../../SelectedWallet'
+import {useBalance, useLockedAmount} from '../../../yoroi-wallets/hooks'
+import {TokenId, YoroiTarget} from '../../../yoroi-wallets/types'
+import {Quantities} from '../../../yoroi-wallets/utils'
+import {useSend} from './SendContext'
+
+export const useTokenQuantities = (tokenId: TokenId) => {
+  const wallet = useSelectedWallet()
+  const {targets, selectedTargetIndex} = useSend()
+  const initialQuantity = targets[selectedTargetIndex].entry.amounts[tokenId] ?? Quantities.zero()
```

Comment:
Amounts.getAmountt

Reply:
i'd like to see all of this logic stripped from react
i want to be able to make these calculations in node

Reply:
id also like to avoid calculating single quantities and multiple quantities in the same scope
so, use `Amounts.diff(amountsA, amountsB)` instead of `Quantities.diff(amountA.quantity, amountB.quantity)

Reply:
so, it consumes amounts, subtracts amounts, returns new amounts